src/session-adapters/internalAdapter.ts

```diff
@@ -1,54 +1,148 @@
 import http from 'http'
 import { Adapter, MaybePromise } from './generalAdapter'
+import {
+  CallbackCookieData,
+  callbackCookieName,
+} from '../storyblok-auth-api/handle-requests/callbackCookie'
+import {
+  expireCookie,
+  getCookie,
+  setCookie,
+  signData,
+  verifyData,
+} from '../utils'
+import { AppSession } from '../session/types'
+import { AuthHandlerParams } from '../storyblok-auth-api'
 
 export type InternalAdapter = {
-  getItem: (key: string) => MaybePromise<string | object | undefined>
-  setItem: (params: {
-    key: string
-    value: string | object
-  }) => MaybePromise<void>
-  removeItem: (key: string) => MaybePromise<void>
-  hasItem: (key: string) => MaybePromise<boolean>
+  // session
+  getSession: (params: {
+    spaceId: string
+    userId: string
+  }) => MaybePromise<AppSession | undefined>
+
+  setSession: (params: {
+    spaceId: string
+    userId: string
+    session: AppSession
+  }) => MaybePromise<boolean>
+
+  removeSession: (params: {
+    spaceId: string
+    userId: string
+  }) => MaybePromise<boolean>
+
+  hasSession: (params: {
+    spaceId: string
+    userId: string
+  }) => MaybePromise<boolean>
+
+  // callback data
+  setCallbackData: (data: CallbackCookieData) => MaybePromise<boolean>
+
+  getCallbackData: () => MaybePromise<CallbackCookieData | undefined>
+
+  removeCallbackData: () => MaybePromise<boolean>
 }
 
 type CreateInternalAdapter = ({
+  params,
   req,
   res,
   adapter,
 }: {
+  params: Pick<AuthHandlerParams, 'clientId' | 'clientSecret' | 'cookieName'>
   req: http.IncomingMessage
   res: http.ServerResponse
   adapter: Adapter
 }) => InternalAdapter
 
 export const createInternalAdapter: CreateInternalAdapter = ({
+  params,
   req,
   res,
   adapter,
-}) => ({
-  getItem: (key) =>
-    adapter.getItem({
-      req,
-      res,
-      key,
-    }),
-  setItem: ({ key, value }) =>
-    adapter.setItem({
-      req,
-      res,
-      key,
-      value,
-    }),
-  hasItem: (key) =>
-    adapter.hasItem({
-      req,
-      res,
-      key,
-    }),
-  removeItem: (key) =>
-    adapter.removeItem({
-      req,
-      res,
-      key,
-    }),
-})
+}) => {
+  const sessionKey = params.cookieName || 'sb.auth'
```

Comment:
do you think it makes more sense to move this to the cookie adapter?

Reply:
I think the logic itself belongs here. Actually that `cookieName` parameter could be renamed like `params.sessionKey`, but it's a breaking change. But... we already have a breaking change in this PR, so we can just make that change and bump the major version? 🤔

Reply:
I was thinking that the argument for moving it to cookieAdapter would be because in a db we would not need it probably. So it is something more related to the cookieStorage.